### DISCUSSION TITLE
go build Windows version from Linux

### DISCUSSION THREAD

**Discussion by GrandeTartarie** [AllenDang/giu#405] [Q&A] [answered]
### Some info:
```
go version go1.17.2 linux/amd64
Windows 10 Pro 21H1 19043.1200
```

### Use this to build:
`CGO_ENABLED=1 GOOS=windows CC=x86_64-w64-mingw32-gcc CXX=x86_64-w64-mingw32-g++ go build -ldflags "-s -w -H=windowsgui -extldflags=-static" -p 4 -v ./cmd/app`

### Got binary:
![image](https://user-images.githubusercontent.com/84810684/139496740-a34338b4-d194-4dbd-9582-4e6510b24cc6.png)

### Try to run:
![Screenshot_1](https://user-images.githubusercontent.com/84810684/139496971-0320c030-b3fa-403c-b81d-a6bb720f6aeb.png)

But **nothing** is happened when I've tried to run it. Nothing

### From CMD:
![Screenshot_2](https://user-images.githubusercontent.com/84810684/139497204-187ff86a-4d70-41bc-bfba-6dc909be74be.png)

The same binary for Linux is two times bigger(15MB) than **.exe**

**Comment by GrandeTartarie** [comment]
Seems like OpenGL required. I've skipped it earlier. Does anyone know about native gui lib to make UI apps?

**Reply by AllenDang** [reply]
Nop, there is no cross-platform GUI lib to create native gui for golang, not even Qt.

**Reply by AllenDang** [reply]
And I don't think there will be one in future. Qt and Flutter also use their own drawing engine to create UI. Native is very complex and hard to maintain.

**Comment by cristianadrielbraun** [comment]
If you have a VM with no OpenGL support, install this:
https://github.com/pal1000/mesa-dist-win
the "mesa3d-xxxx-release-msvc.7z" allways worked to me.
I was struggling trying to make GUI programs in Go because I use a lot of Hyper-V and XenServer VMs, and both of them are limited to OpenGL 1.1 I believe. That driver allows to run OpenGL 3+ software rendered

PS: More info about the Mesa drivers: https://en.wikipedia.org/wiki/Mesa_(computer_graphics)

**Comment by gucio321** [comment, marked as answer]
@GrandeTartarie any updates here?
Last days I was fighting with this cross-platform builds and got it working.
take a look here: https://github.com/AllenDang/giu/issues/449
